src/core_modules/capture-core/components/WidgetEventEdit/DataEntry/withDeleteButton.js

```diff
@@ -70,14 +71,14 @@ const getDeleteButton = (InnerComponent: React.ComponentType<any>) =>
         )
 
         render() {
-            const { onDelete, ...passOnProps } = this.props;
+            const { onDelete, hasDeleteButton, ...passOnProps } = this.props;
             const hasWriteAccess = this.props.formFoundation.access.data.write;
             return (
                 // $FlowFixMe[cannot-spread-inexact] automated comment
                 <InnerComponent
                     innerRef={(innerInstance) => { this.innerInstance = innerInstance; }}
                     // $FlowFixMe[extra-arg] automated comment
-                    deleteButton={this.renderDeleteButton(hasWriteAccess)}
+                    deleteButton={hasDeleteButton ? this.renderDeleteButton(hasWriteAccess) : null}
```

Comment:
I think we can clean up the `hasWriteAccess` here because it's not used in the method, maybe pass the `hasDeleteButton` as parameter?